### PR TITLE
[IOPLT-243] adds mergeFields function

### PR DESCRIPTION
**Pull request by StefanoFrontini**
The function takes as input a Javascript object T, a list of fields of type [keyof T], another string newField , a separator and returns the input with a newField containing the values joined with the separator. The separator may not be mandatory, the default value is the comma. 

Example:

Input:
{
   foo: "foo",
   bar: 1,
   baz: "hello"
},
["foo", "baz"],
"mergedField",
-
----------------------------
Output : 
{
   mergedField: "foo-hello",
   bar: 1
}

#### Motivation and Context

Formatting data coming from data sources.

#### How Has This Been Tested?

Unit tests. See mergeFields.test.ts

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ X] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/formatter/mergeFields.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import * as A from "fp-ts/Array";
+import * as E from "fp-ts/Either";
+import { has } from "fp-ts/Record";
+import { pipe } from "fp-ts/function";
+export const mergeFields = <T extends Record<string, unknown>>(
+  input: T,
+  fields: ReadonlyArray<keyof T>,
+  newField: string,
+  separator: string = ","
```
Comment (AleDore) [inline src/formatter/mergeFields.ts:9]:
```suggestion
  separator?: string = ","
```

File: src/formatter/mergeFields.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  pipe(
+    A.traverse(E.Applicative)((field: keyof T) =>
+      has(String(field), input)
+        ? E.right(String(input[field]))
+        : E.left(new Error("Not all fields are present"))
+    )(Array.from(fields)),
+    E.map(fieldValues => ({
+      ...input,
+      [newField]: fieldValues.join(separator)
+    }))
```
Comment (AleDore) [inline src/formatter/mergeFields.ts:20]: more simple:
```suggestion
  pipe(
    fields.map(field => input[field]),
    fieldValues => ({
      ...input,
      [newField]: fieldValues.join(separator)
    })
  );
```

File: src/formatter/__test__/mergeFields.test.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  it("should output an empty string if no fields are provided", () => {
+    const res = mergeFields(inputData, [], "mergedField", "-");
+    expect(res).toEqual({
+      bar: 1,
+      baz: "hello",
+      foo: "Foo",
+      mergedField: ""
+    });
+  });
+  it("should output an empty string if a missing field is provided", () => {
```
Comment (AleDore) [inline src/formatter/__test__/mergeFields.test.ts:28]: What do you think about this behavior @drmarro @CrisTofani ? Do you prefer to return an empty string or returning the same input with no new field added to data stream?

Comment (CrisTofani) [reply src/formatter/__test__/mergeFields.test.ts:28]: I think that if no edit happens we should know, maybe an empty array would be more representative

**Review by AleDore — APPROVED** [review]

**Pull request merged**